packages/hdwallet-metamask/src/metamask.ts

```diff
@@ -261,18 +277,30 @@ export class MetaMaskHDWallet implements core.HDWallet, core.ETHWallet {
     }
   }
 
+  public async ethAddChain(params: AddEthereumChainParameter): Promise<void> {
+    // at this point, we know that we're in the context of a valid MetaMask provider
+    await this.provider.request({ method: "wallet_addEthereumChain", params: [params] });
+  }
+
   public async ethSwitchChain(chainId: number): Promise<void> {
     const hexChainId = ethers.utils.hexValue(chainId);
     try {
       // at this point, we know that we're in the context of a valid MetaMask provider
       await this.provider.request({ method: "wallet_switchEthereumChain", params: [{ chainId: hexChainId }] });
     } catch (e: any) {
-      const error: core.SerializedEthereumRpcError = e;
-      console.error(error);
-      if (error.code === 4902) {
-        // TODO: EVM Chains Milestone
-        // We will need to pass chainName and rpcUrls, which we don't have yet, to add a chain to MetaMask.
+      const error = serializeError(e);
+      if (error.code === -32603) {
```

Comment:
Same thought here about the magic number error code.

Reply:
Added in eb3093a225d13b9e439f28cf3addec817661cbd8